code/modules/1713/lantern.dm

```diff
@@ -237,4 +237,44 @@
 	item_state = "modern_flashlight"
 	on_state = "modernlight_on"
 	off_state = "modernlight_off"
-	slot_flags = SLOT_BELT|SLOT_MASK
\ No newline at end of file
+	slot_flags = SLOT_BELT|SLOT_MASK
+
+/obj/item/flashlight/militarylight
+	unlimited = TRUE
+	name = "military flashlight"
+	desc = "an electrical military flashlight."
+	icon_state = "militarylight_off"
+	item_state = "militarylight"
+	on_state = "militarylight_on"
+	off_state = "militarylight_off"
+	slot_flags = SLOT_BELT|SLOT_MASK
+	var/switchlens = FALSE
+
+
+/* please can someone make this work?
+/obj/item/flashlight/militarylight/verb/switchlens()
+	set category = null
+	set src in usr
+	set name = "Switch lens"
+	if (type != /obj/item/flashlight/militarylight)
+		return
+	else
+		if (!switchlens)
+			light_power = 2
+			light_color = "#e58775"
+			on_state = "militarylight_on_red"
+			switchlens = TRUE
+			update_icon()
+		else if (!switchlens)
```

Comment:
yo have 2 exact same conditions
instead create a var like "millight_state = 1" then your 3 if conditions could be for 1 2 3 and switch to the next when needing done.